pkg/providers/subnet/subnet.go

```diff
@@ -96,20 +104,24 @@ func (p *DefaultProvider) List(ctx context.Context, nodeClass *v1beta1.EC2NodeCl
 		if err != nil {
 			return nil, fmt.Errorf("describing subnets %s, %w", pretty.Concise(filters), err)
 		}
+		availableZones := map[string]string{}
 		for i := range output.Subnets {
 			subnets[lo.FromPtr(output.Subnets[i].SubnetId)] = output.Subnets[i]
 			p.availableIPAddressCache.SetDefault(lo.FromPtr(output.Subnets[i].SubnetId), lo.FromPtr(output.Subnets[i].AvailableIpAddressCount))
 			p.associatePublicIPAddressCache.SetDefault(lo.FromPtr(output.Subnets[i].SubnetId), lo.FromPtr(output.Subnets[i].MapPublicIpOnLaunch))
 			// subnets can be leaked here, if a subnets is never called received from ec2
 			// we are accepting it for now, as this will be an insignificant amount of memory
 			delete(p.inflightIPs, lo.FromPtr(output.Subnets[i].SubnetId)) // remove any previously tracked IP addresses since we just refreshed from EC2
+			// add zone name to ID comparison as subnets are discovered
+			availableZones[aws.StringValue(output.Subnets[i].AvailabilityZone)] = aws.StringValue(output.Subnets[i].AvailabilityZoneId)
 		}
+		p.zoneInfo = availableZones
```

Comment:
Any reason not to update the data directly rather than creating a new map and assigning it?

Reply:
Found that this is actually wrong because we overwrite the global provider map each time we list a set of filtered subnets. In reality, this should be merged each time.